src/box/index.cc

```diff
@@ -806,6 +806,31 @@ index_delete(struct index *index)
 	index_def_delete(def);
 }
 
+int
+index_replace_with_txm(struct index *index, struct tuple *old_tuple,
+		       struct tuple *new_tuple, enum dup_replace_mode mode,
+		       struct tuple **result, struct tuple **successor)
+{
+	memtx_tx_story_gc();
+	if (new_tuple == NULL)
+		return memtx_tx_history_add_delete(index, old_tuple, result);
+	if (index->vtab->replace(index, NULL, new_tuple, DUP_REPLACE_OR_INSERT,
+				 result, successor) != 0)
+		return -1;
+	if (memtx_tx_history_add_insert(index, old_tuple, new_tuple, mode,
```

Comment:
Looks like this function is memtx-specific. I'd make it local to `memtx_space.cc`.

Reply:
Actually, AFAIC, the whole index manipulation is memtx-specific — vinyl goes through it's own path:
https://github.com/tarantool/tarantool/blob/921a07177521cfdea7c59ec84a78b19be7ef44ff/src/box/vinyl.c#L4653-L4668

> I'd make it local to `memtx_space.cc`.

I don't understand what you mean, could you please elaborate?

Reply:
`src/box/index.cc` shouldn't depend on memtx. I'd move this function to `src/box/memtx_space.cc`, because it's the only file where it's used.

Reply:
Refactored.